esphome/components/lvgl/widgets/arc.py

```diff
@@ -59,16 +72,33 @@ def __init__(self):
 
     async def to_code(self, w: Widget, config):
         if CONF_MIN_VALUE in config:
-            max_value = await lv_int.process(config[CONF_MAX_VALUE])
+            max_value = lv.arc_get_max_value(w.obj)
             min_value = await lv_int.process(config[CONF_MIN_VALUE])
             lv.arc_set_range(w.obj, min_value, max_value)
+
+        if CONF_MAX_VALUE in config:
+            max_value = await lv_int.process(config[CONF_MAX_VALUE])
+            min_value = lv.arc_get_min_value(w.obj)
+            lv.arc_set_range(w.obj, min_value, max_value)
+
+        if CONF_START_ANGLE in config:
             start = await lv_angle_degrees.process(config[CONF_START_ANGLE])
+            lv.arc_set_bg_start_angle(w.obj, start)
+
+        if CONF_END_ANGLE in config:
             end = await lv_angle_degrees.process(config[CONF_END_ANGLE])
+            lv.arc_set_bg_end_angle(w.obj, end)
+
+        if CONF_ROTATION in config:
             rotation = await lv_angle_degrees.process(config[CONF_ROTATION])
-            lv.arc_set_bg_angles(w.obj, start, end)
             lv.arc_set_rotation(w.obj, rotation)
+
+        if CONF_MODE in config:
             lv.arc_set_mode(w.obj, literal(config[CONF_MODE]))
-            lv.arc_set_change_rate(w.obj, config[CONF_CHANGE_RATE])
+
+        if CONF_CHANGE_RATE in config:
+            change_rate = await lv_positive_int.process(config[CONF_CHANGE_RATE])
+            lv.arc_set_change_rate(w.obj, change_rate)
```

Comment:
This and similar sequences can be simplified with the `set_property()` method. It does nothing if the value is None.

```suggestion
        w.set_property(CONF_CHANGE_RATE, await lv_positive_int.process(config.get(CONF_CHANGE_RATE)))
```